service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -323,8 +332,20 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
         new RunSetStateResponse().runSetId(runSetId).runs(runStateResponseList).state(runSetState);
 
     captureResponseMetrics(response);
-    bardService.logRunSetEvent(request, userToken);
-
+    if (bardServerConfiguration.enabled()) {
+      List<String> workflowIds =
+          runStateResponseList.stream()
```

Comment:
This strikes me as odd - that we have to update the API types just to get a value internally in the system so we can log it.

I believe this becomes a little more sane in https://github.com/DataBiosphere/cbas/pull/281 which splits the Run-creation away from the submitting. So you might have the information you need more easily available after that.

Reply:
Oh I'll take a look!

Reply:
I merged in those changes and I still have the same issue. Since `buildInputsAndSubmitRun` returns `List<RunStateResponse>`, I thought it made sense to include the engine id in `RunStateResponse`, but an alternative could be to make it return a new object type with `runStateResponse`` and `workflowIds` fields. Is there a reason why you don't want it to be in the run state response?